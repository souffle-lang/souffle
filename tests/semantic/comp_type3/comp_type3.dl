.comp X<C> {

  .comp C1 {
     .type T = number
     A(1).
  } 

  .comp C2 { 
     .type T
     A("a").
  } 

  .decl A(x:T) printsize

  .init select = C
}

.init myC=X<C1> 
// Alternative .init myC=X<C2> 
