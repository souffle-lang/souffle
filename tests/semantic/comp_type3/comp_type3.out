myC_A	1
